tuna.go

```diff
@@ -476,29 +491,49 @@ func (c *Common) CreateServerConn(force bool) error {
 				return err
 			}
 
-			subscribersCount, err := c.Wallet.GetSubscribersCount(topic)
-			if err != nil {
-				return err
-			}
-			if subscribersCount == 0 {
-				return errors.New("there is no service providers for " + c.Service.Name)
-			}
+			var allSubscribers []string
+			var filterSubs filterSubscribers
+			var subscriberRaw map[string]string
 
-			offset := rand.Intn(subscribersCount/getSubscribersBatchSize + 1)
-			subscribers, err := c.Wallet.GetSubscribers(topic, offset*getSubscribersBatchSize, getSubscribersBatchSize, true, false)
-			if err != nil {
-				return err
-			}
+			if c.ServiceInfo.NknFilter != nil && len(c.ServiceInfo.NknFilter.Allow) > 0 { // nkn client filter
+				nknFilterLength := len(c.ServiceInfo.NknFilter.Allow)
+				subscriberRaw = make(map[string]string, nknFilterLength)
+				allSubscribers = make([]string, 0, nknFilterLength)
+				for _, aFilter := range c.ServiceInfo.NknFilter.Allow {
+					subscription, err := c.Wallet.GetSubscription(topic, aFilter.Address)
+					if err != nil {
```

Comment:
We should print and `continue` here instead of return. If we get error for all addresses in whitelist, then we should return error